### PR TITLE
Updated newznab API for movies

### PR DESCRIPTION
**Pull request by the-darkvoid**
Changes to support Radarr working with Spotweb:

- Updated movie search with “q” and “imdbid”
- Added latest movie listing (required for Radarr)
- Added “q” support for movies

### CODE REVIEW

File: lib/page/SpotPage_newznabapi.php [hunk 1]
```diff
@@ -200,48 +200,67 @@ function search($outputtype) {
...
-
-				return ;
-			} elseif (!preg_match('/^[0-9]{1,8}$/', $this->_params['imdbid'])) {
-				$this->showApiError(201);
+			/*
+			* Query by IMDB id
+			*/
+			if (!empty($this->_params['imdbid'])) {
+				# validate input
+				if ($this->_params['imdbid'] == "") {
```
Comment (mesa57) [inline lib/page/SpotPage_newznabapi.php]: This makes no sense, you just tested is imdbid is not empty

File: lib/page/SpotPage_newznabapi.php [hunk 2]
```diff
@@ -200,48 +200,67 @@ function search($outputtype) {
...
+					$this->showApiError(200);
+
+					return ;
+				} elseif (!preg_match('/^[0-9]{1,8}$/', $this->_params['imdbid'])) {
+					$this->showApiError(201);
+
```
Comment (mesa57) [inline lib/page/SpotPage_newznabapi.php]: Superfluous empty line

Comment (onedr0p) [inline lib/page/SpotPage_newznabapi.php]: just like my soul

**Comment by the-darkvoid** [comment]
In the original code imdb db is implicitly expected. However the newznab API (as well as the queries used by Radarr) do not follow that.

Hence radarr fails to add spotweb as an indexer. Note that for tv-search this works fine.

| Case | URL                                                                            | Expected                                 | Existing | Revised | Remarks                                                                         |
|------|--------------------------------------------------------------------------------|------------------------------------------|----------|---------|---------------------------------------------------------------------------------|
| 1    | `/spotweb/api?t=movie&imdbid=1386697&apikey=xxx&extended=1`                    | Movie as per IMDB ID                     | Pass     | Pass    |                                                                                 |
| 2    | `/spotweb/api?t=movie&imdbid=xyz&apikey=xxx&extended=1`                        | Invalid IMDB Message                     | Pass     | Pass    |                                                                                 |
| 3    | `/spotweb/api?t=movie&cat=2000,2010&extended=1&apikey=xxx&offset=0&limit=50`   | List of first 50 movies  in categories   | Fail     | Pass    | Existing code expects IMDB ID, new code only runs IMDB logic if it is non-empty |
| 4    | `/spotweb/api?t=movie&extended=1&apikey=xxx&offset=0&limit=50`                 | List of first 50 movies in category 2000 | Fail     | Pass    | See 3 and default category is not defined as 2000 for movies                    |
| 5    | `/spotweb/api?t=movie&extended=1&apikey=xxx&offset=0&limit=50&q=Suicide+Squad` | Free query in movies                     | Fail     | Pass    | See 3 and 'q' is not supported in existing code                                 |'

File: lib/page/SpotPage_newznabapi.php (hunk 1 above)
Comment (the-darkvoid) [inline lib/page/SpotPage_newznabapi.php]: The test is to validate if imdb logic should run or not, see latest pull comment with test cases

Comment (mesa57) [inline lib/page/SpotPage_newznabapi.php]: I understand you're remark, but I think this piece of code is obsolete : 
```
				# validate input
				if ($this->_params['imdbid'] == "") {
					$this->showApiError(200);

					return ;
```

Comment (the-darkvoid) [inline lib/page/SpotPage_newznabapi.php]: True, that is the original logic though which was in place.
It discerns between api error 200 (parameter missing) and api error 201 (parameter incorrect).

Happy to remove the line in question, I just tried to leave the existing logic intact where possible.

Comment (mesa57) [inline lib/page/SpotPage_newznabapi.php]: It has no sense to leave it in as the code is never reached any more.
So please remove that dead code.

Comment (the-darkvoid) [inline lib/page/SpotPage_newznabapi.php]: Change made as requested.

**Pull request merged**